crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -0,0 +1,485 @@
+                Operation::Phi { block_args, .. } => {
+                    if block_kind == BlockType::IfJoin {
+                        assert_eq!(block_args.len(), 2);
+                        let ins2 = ctx.get_mut_instruction(*i);
+                        ins2.operation = Operation::Cond {
+                            condition: assumption.condition,
+                            val_true: block_args[0].0,
+                            val_false: block_args[1].0,
+                        }
+                    }
+                }
+
+                Operation::Store { array_id, index, value } => {
+                    if !ins.operation.is_dummy_store() && ass_value != ctx.one() {
+                        let load = Operation::Load { array_id: *array_id, index: *index };
+                        let e_type = ctx.mem[*array_id].element_type;
+                        let dummy =
+                            ctx.add_instruction(Instruction::new(load, e_type, Some(block)));
+                        let pos = ctx[block]
+                            .instructions
+                            .iter()
+                            .position(|value| *value == ins.id)
+                            .unwrap();
+                        ctx[block].instructions.insert(pos - 1, dummy);
+                        let operation = Operation::Cond {
+                            condition: ass_value,
+                            val_true: *value,
+                            val_false: dummy,
+                        };
+                        let cond =
+                            ctx.add_instruction(Instruction::new(operation, e_type, Some(block)));
+                        ctx[block].instructions.insert(pos, cond);
+                        //store the conditional value
+                        let ins2 = ctx.get_mut_instruction(*i);
+                        ins2.operation =
+                            Operation::Store { array_id: *array_id, index: *index, value: cond };
+                    }
+                }
+                Operation::Intrinsic(_, _) => {
+                    if ass_value != ctx.one() {
+                        if let ObjectType::Pointer(a) = ins.res_type {
+                            let array = &ctx.mem[a].clone();
+                            let name = array.name.to_string() + "_dup";
+                            ctx.new_array(&name, array.element_type, array.len, None);
+                            let array_dup = ctx.mem.last_id();
+                            let ins2 = ctx.get_mut_instruction(*i);
+                            ins2.res_type = ObjectType::Pointer(array_dup);
+                            let mut stack = StackFrame::new(block);
+                            ctx.memcpy_inline(
+                                ins.res_type,
+                                ObjectType::Pointer(array_dup),
+                                &mut stack,
+                            );
+                            todo!();
```

Comment:
For todos, please leave a note in the block as to what needs to be done. For example: `todo!("feature X is not implemented")


If you are also not going to remove this todo now, lets open an issue to keep track of it

Reply:
Done
n.b. I did not create an issue because I am working on it right now and will create a new PR for this feature.